pulp_ansible/app/migrations/0042_collectionversion_sha256.py

```diff
@@ -0,0 +1,103 @@
+# Generated by Django 3.2.14 on 2022-07-15 22:51
+
+from django.db import migrations, models
+from git import GitCommandError, Repo
+from tempfile import TemporaryDirectory
+from uuid import uuid4
+
+import subprocess
+
+
+def add_sha256_to_current_models(apps, schema_editor):
+    """Adds the sha256 to current CollectionVersion models."""
+    CollectionVersion = apps.get_model('ansible', 'CollectionVersion')
+    ContentArtifact = apps.get_model('core', 'ContentArtifact')
+    RemoteArtifact = apps.get_model('core', 'RemoteArtifact')
+    Artifact = apps.get_model('core', 'Artifact')
+    collection_bulk = {}
```

Comment:
Getting it to work is the first prio of course. But in the end, we want this migration to work in batches.
On a really large installation, one by one is way to slow, but all in one will blow up the ram.

Reply:
After asking if anyone uses GitRemotes (which can create on-demand Collections), it seems it is not widespread, so I simplified the migration to only migrate collections that have artifacts in the system. If on-demand collections are present an exception is raised with the collections that failed.